crossdock/client/echo.go

```diff
@@ -0,0 +1,81 @@
+package client
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"encoding/base64"
+
+	"golang.org/x/net/context"
+
+	"github.com/yarpc/yarpc-go"
+	"github.com/yarpc/yarpc-go/encoding/json"
+	"github.com/yarpc/yarpc-go/transport"
+	"github.com/yarpc/yarpc-go/transport/http"
+)
+
+// EchoRequest contains a message to echo
+type EchoRequest struct {
```

Comment:
perhaps just reach into the server package and use `EchoRequest` and `EchoResponse` from there... 

Although ideally these are implemented entirely seperate...

@kriskowal what do you think?